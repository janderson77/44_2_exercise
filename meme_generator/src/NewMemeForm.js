import React, {useState} from 'react'
import {v4 as uuid} from 'uuid'

const NewMemeForm = ({addMeme}) => {
    const [fields, setFields] = useState({
        topText: '',
        bottomText: '',
        url: ''
    })

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFields(fields => ({
            ...fields,
            [name]: value
        }));
    };

    const makeMeme = (e) => {
        e.preventDefault();
        addMeme({...fields, id: uuid()});
        setFields({
            topText: '',
            bottomText: '',
            url: ''
        })
    }

    return(
        <div>
            <form onSubmit={makeMeme}>
                <div className="form-group">
                    <label>Top Text</label>
                    <input 
                        name="topText"
                        className="form-control"
                        value={fields.topText}
                        onChange={handleChange}
                    />
                </div>
                <div className="form-group">
                    <label>Bottom Text</label>
                    <input 
                        name="bottomText"
                        className="form-control"
                        value={fields.bottomText}
                        onChange={handleChange}
                    />
                </div>
                <div className="form-group">
                    <label>Image URL</label>
                    <input 
                        name="url"
                        className="form-control"
                        value={fields.url}
                        onChange={handleChange}
                    />
                </div>
                <button
                    type="submit"
                    className="btn btn-primary"
                    onSubmit={makeMeme}
                >Submit</button>
            </form>
        </div>
        
    )
}

export default NewMemeForm