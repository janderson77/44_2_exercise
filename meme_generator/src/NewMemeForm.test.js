import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewMemeForm from "./NewMemeForm";

let memeForm;

describe('NewMemeForm basic stuff', () => {
    beforeEach(() => {
        memeForm = render(<NewMemeForm />)
    })

    it('renders without crashing', () => {
        expect(memeForm).toBeTruthy()
    })
    it('matches snapshot', () => {
        expect(memeForm.asFragment()).toMatchSnapshot()
    })
})

it('updates state with topText input', () => {
    const mock = jest.fn()
    const {getByText} = render(<NewMemeForm addMeme={mock} />)

    fireEvent.click(getByText('Submit'))

    expect(mock).toHaveBeenCalled()
})