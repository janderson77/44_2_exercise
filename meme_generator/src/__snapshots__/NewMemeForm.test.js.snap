// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMemeForm basic stuff matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div
        class="form-group"
      >
        <label>
          Top Text
        </label>
        <input
          class="form-control"
          name="topText"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Bottom Text
        </label>
        <input
          class="form-control"
          name="bottomText"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Image URL
        </label>
        <input
          class="form-control"
          name="url"
          value=""
        />
      </div>
      <button
        class="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
